packages/core/src/lib/metadata-delegate.ts

```diff
@@ -0,0 +1,70 @@
+import assign from './es-utils/assign'
+
+interface MetadataDelegate {
+  add: (state: { [key: string]: any }, section: string, keyOrObj?: object | string, maybeVal?: any) => any
+  get: (state: { [key: string]: any }, section: string, key?: string | undefined) => any
+  clear: (state: { [key: string]: any }, section: string, key?: string | undefined) => any
```

Comment:
the question mark in `key?: string` also allows for `undefined` so that's probably fine as it was.